### ISSUE TITLE
Not really local?

### ISSUE THREAD

**Issue by flatsiedatsie** [Bleznudd/avocet#1]
The description calls this a local voice assistant, so I was very interested. But it seems to rely on a cloud service for intent recognition?

(I'm the developer of the Voco addon)

**Comment by Bleznudd** [comment]
It does rely on a cloud service, but only for token verification.

Picovoice lets you use one token for max 3 devices, so some data is sent to their servers to recognize the device and count it as one. Once that is done, there's no need of an internet access any more.

You can try this by yourself launching the gateway along with the addon, and then disabling wifi.
This way you will lose the voice feedback though, which as of now is provided by gtts

Thank you for opening the first issue! 😄

**Comment by flatsiedatsie** [comment]
gtts, as in Google Text To Speech? Doesn't that imply a second cloud dependency?

I hope I don't sound puritanical, but I ask because I value that the word 'local' keeps meaning 'no cloud connection whatsever required', as it does with Voco.

**Comment by Bleznudd** [comment]
I see a difference between **local** and **local only**.

A software being able to run locally is a software that doesn't _strictly_ require an internet connection to be used (as Avocet, since you can turn on and off things without being online), but that can take advantage of it to do more things (as Avocet, using it to give you a voice confirmation).
On the other hand, a local only software doesn't care if it's connected or not, as it will behave in the same way in both conditions (as, I guess, Voco).

Anyway, the focus here was on building a lightweight and polyglot voice assistant (like no other I'm aware of) while keeping the audio from being sent to a cloud server, for privacy reasons. The local adjective is being used because of this; because the intent recognition and the subsequent action, does happen locally. 
Sure it's nice to have a voice feedback stating that the action has been executed, but if I want to shut down the connection for whatever reason, I can still use (most of) the assistant without the feedback and live happily with it.

Also keep in mind that as I wrote in the release comment 

> The software is now barely more than a working prototype

Neither the intent recognition nor the text to speech software are going to last, since none of them satisfies me fully.
It's highly likely that a future release won't need an internet connection to give voice feedback.
